Xledger.Collections/Memory/SizedMemoryOwner.cs

```diff
@@ -0,0 +1,11 @@
+﻿using System.Buffers;
+
+namespace Xledger.Collections.Memory;
+
+sealed class SizedMemoryOwner<T>(IMemoryOwner<T> memoryOwner, int capacity) : IMemoryOwner<T> {
+    public Memory<T> Memory { get; } = memoryOwner.Memory.Slice(0, capacity);
+
+    public void Dispose() {
```

Comment:
Here I think we should probably have a finalizer, and a 'IsDisposed' check, like this one from Trell:

![image](https://github.com/user-attachments/assets/9f66ad3e-a7a4-47ac-a073-04c50c6011dc)